src/asynqp/channel.py

```diff
@@ -37,7 +37,7 @@ def __init__(self, id, synchroniser, sender, basic_return_consumer, queue_factor
         self.reader = reader
 
     @asyncio.coroutine
-    def declare_exchange(self, name, type, *, durable=True, auto_delete=False, internal=False):
+    def declare_exchange(self, name, type, *, durable=True, auto_delete=False, internal=False, arguments={}):
```

Comment:
That's a good argument against `**kwargs`, you've convinced me :wink:

I still would prefer an immutable default parameter because it's much more maintainable. All it takes is one mistaken change in the future and it'll blow up on you.
What I usually do is set the default to `None` and then check for that in the body of the method:

```python
def declare_exchange(self, ..., arguments=None):
    if arguments is None:
        arguments = {}
    ...
```